backend/services/implementations/shiftSignupService.ts

```diff
@@ -0,0 +1,87 @@
+import { PrismaClient, Signup, SignupStatus } from "@prisma/client";
+
+import IShiftSignupService from "../interfaces/shiftSignupService";
+import {
+  CreateShiftSignupDTO,
+  ShiftSignupDTO,
+  ShiftSignupResponseDTO,
+  UpdateShiftSignupRequestDTO,
+} from "../../types";
+import logger from "../../utilities/logger";
+
+const prisma = new PrismaClient();
+
+const Logger = logger(__filename);
+
+class ShiftSignupService implements IShiftSignupService {
+  convertSignupToDTO = (signup: Signup): ShiftSignupDTO => {
+    return {
+      ...signup,
+      shiftId: String(signup.shiftId),
+      userId: String(signup.userId),
+    };
+  };
+
+  async getShiftSignupsForUser(
+    userId: string,
+  ): Promise<ShiftSignupResponseDTO[]> {
+    try {
+      const shiftSignups = await prisma.signup.findMany({
+        where: { userId: Number(userId) },
+      });
+      return shiftSignups.map((signup) => this.convertSignupToDTO(signup));
+    } catch (error) {
+      Logger.error(`Failed to shift signups. Reason = ${error.message}`);
+      throw error;
+    }
+  }
+
+  async createShiftSignups(
+    shiftSignups: CreateShiftSignupDTO[],
+  ): Promise<ShiftSignupResponseDTO[]> {
+    try {
+      const newShiftSignups = await prisma.$transaction(
+        shiftSignups?.map((shiftSignup) =>
```

Comment:
```suggestion
        shiftSignups.map((shiftSignup) =>
```

Do we need optional chaining here? `shiftSignups` should be non-null.

Reply:
Hmmm, I don't think so. I just like adding it for safety

Reply:
Hmm ok, not a big deal here but I'd rather avoid optional chaining when not necessary and have it fail in an obvious manner if an unexpected null value is passed

Reply:
Hmm I see your point, I'll change it